skimage/filters/thresholding.py

```diff
@@ -927,6 +927,12 @@ def threshold_triangle(image, nbins=256):
     >>> thresh = threshold_triangle(image)
     >>> binary = image > thresh
     """
+    # Check if the image has more than one intensity value; if not, return that
+    # value
+    if image is not None:
+        first_pixel = image.reshape(-1)[0]
+        if np.all(image == first_pixel):
+            return first_pixel
```

Comment:
I agree with the goal, that `threshold_triangle` should return the intensity value if the input is uniform. However, this check `image == first_pixel` might be unnecessarily expensive, considering that I expect the input not to be uniform most of the time.

Instead how about we insert the following
```python
# after arg_low_level, arg_high_level = np.flatnonzero(hist)[[0, -1]]
if arg_low_level == arg_high_level:
    return image.ravel()[0]
```

Reply:
Great suggest. I had explored this solution prior. I find that it returns a random float number for `uniform_image.dtype == 'float'` instead of the expected value of zero. e.g
```py
zero_image2 = np.zeros((10, 10))
threshold_triangle(zero_image2)
0.001953125
```

Reply:
How about a mix of both approaches?
```py
if arg_low_level == arg_high_level:
    return image.reshape(-1)[0]
```

Reply:
Great catch! All 0 values get sorted into the bin [0,1/256), which has it's center at 0.001953125. I am a bit undecided, whether we should return 0 or 0.001953125. 0 definitely seems more intuitive at a first glance, but one could argue that 0.001953125 is technically more consistent with the behavior for non-uniform images. Though, this algorithm clearly assumes that images are not uniform. So, let's go with the more intuitive option, and return 0. 

Nitpick:
```python
image.ravel()[0]
```
is a bit faster (on my machine) and IMO more expressive than the `reshape(-1)` based option.